roles/cockpit-ui/tasks/main.yml

```diff
@@ -45,17 +43,68 @@
       {% if openshift_cockpit_deployer_version is defined  %}-p IMAGE_VERSION="{{ openshift_cockpit_deployer_version }}"{% endif %}
       -p OPENSHIFT_OAUTH_PROVIDER_URL="{{ openshift.master.public_api_url }}"
       -p REGISTRY_HOST="{{ docker_registry_route.results[0].spec.host }}"
-      -p COCKPIT_KUBE_URL="https://{{ registry_console_cockpit_kube.results.results[0].spec.host }}"
-      --config={{ openshift_hosted_kubeconfig }}
-      -n default
+      -p COCKPIT_KUBE_URL="https://{{ registry_console_route.results[0].spec.host }}"
+      --config={{ openshift_master_config_dir }}/admin.kubeconfig
+      -n {{ openshift_hosted_registry_console_namespace }}
     register: deploy_registry_console
     changed_when: "'already exists' not in deploy_registry_console.stderr"
     failed_when: "'already exists' not in deploy_registry_console.stderr and deploy_registry_console.rc != 0"
 
-  - name: Delete temp directory
-    file:
-      name: "{{ mktemp.stdout }}"
-      state: absent
-    changed_when: False
-    # XXX: End required for items still using command
+  - name: Retrieve registry-console service for the clusterip
+    oc_service:
+      namespace: "{{ openshift_hosted_registry_console_namespace }}"
+      name: registry-console
+      state: list
+    register: docker_registry_console_service
+
+  - name: Generate self-signed registry-console certificates
+    oc_adm_ca_server_cert:
+      signer_cert: "{{ openshift_master_config_dir }}/ca.crt"
+      signer_key: "{{ openshift_master_config_dir }}/ca.key"
+      signer_serial: "{{ openshift_master_config_dir }}/ca.serial.txt"
+      hostnames:
+      - "{{ docker_registry_console_service.results.clusterip }}"
+      - "{{ registry_console_route.results[0].spec.host }}"
+      cert: "{{ docker_registry_console_cert_path }}"
+      key: "{{ docker_registry_console_key_path }}"
+      expire_days: "{{ openshift_hosted_registry_console_cert_expire_days if openshift_version | oo_version_gte_3_5_or_1_5(openshift.common.deployment_type) | bool else omit }}"
+    when: docker_registry_console_self_signed
+
+  # The certificate file expected inside the registry-console service is
+  # in .pem format that bundles the certificate(s) and the private key.
+  - name: Retrieve certificate files to generate certificate bundle
+    slurp:
+      src: "{{ item }}"
+    with_items: >
+      {%- set files = [ docker_registry_console_cert_path ] %}
+      {%- if docker_registry_console_cacert is defined %}
+      {%-   set _ = files.append(docker_registry_console_cacert) %}
+      {%- endif %}
+      {%- set _ = files.append(docker_registry_console_key_path) %}
+      {{- files -}}
+    register: certificate_files
+
+  - name: Generate certificate bundle
+    copy:
+      content: "{{ certificate_files.results | map(attribute='content') | map('b64decode') | join('') }}"
+      dest: "{{ openshift_master_config_dir }}/named_certificates/registry-console.pem"
```

Comment:
Not a bad idea to put it somewhere temporary. That said, I guess we don't need to keep *any* certificate around once they have been configured either directly on the route (passthrough) or as a secret (mounted as a volume).

That said, I guess you sort of lose some idempotency since you'll need to upload certs in tmpdir/generate bundle/check against what is configured/delete tmpdir.
The fact that the bundle is stored means the task should not be "changed" because the source files are already there and the bundle ends up the same.